### PR TITLE
Add Keychron Q12

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
Added new keyboard Q12 which contains two sub versions.
Thanks!

## Description

- Added ANSI and ISO versions with encoder support.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q12/config.h [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    { B5, B4, B3, A15, A14, A13 }
+#define MATRIX_COL_PINS \
+    { A10, A9, A8, B1, B0, A7, A6, A5, A4, A3, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN }
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION ROW2COL
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
```
Comment (fauxpark) [inline keyboards/keychron/q12/config.h:30]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q12/config.h (hunk 1 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q12/config.h]: Thanks!

File: keyboards/keychron/q12/ansi_encoder/config.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+/* RGB Matrix Configuration */
+#define DRIVER_1_LED_TOTAL 56
+#define DRIVER_2_LED_TOTAL 46
+#define RGB_MATRIX_LED_COUNT (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+
+/* Encoder Configuration */
+#define ENCODERS_PAD_A { A2 }
+#define ENCODERS_PAD_B { C14 }
+#define ENCODER_RESOLUTION 4
```
Comment (drashna) [inline keyboards/keychron/q12/ansi_encoder/config.h:27]: This is the default and shouldn't be defined here. 
```suggestion
```

File: keyboards/keychron/q12/iso_encoder/config.h [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+/* RGB Matrix Configuration */
+#define DRIVER_1_LED_TOTAL 56
+#define DRIVER_2_LED_TOTAL 47
+#define RGB_MATRIX_LED_COUNT (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+
+/* Encoder Configuration */
+#define ENCODERS_PAD_A { A2 }
+#define ENCODERS_PAD_B { C14 }
+#define ENCODER_RESOLUTION 4
```
Comment (drashna) [inline keyboards/keychron/q12/iso_encoder/config.h:27]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q12/iso_encoder/config.h (hunk 3 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q12/iso_encoder/config.h:27]: All right, Thanks!

**Review by drashna — APPROVED** [review]

**Comment by flowchartsman** [comment]
My Q12 arrived today, any chance we can get this merged?

File: keyboards/keychron/q12/ansi_encoder/info.json [hunk 4]
```diff
@@ -0,0 +1,127 @@
+{
+    "keyboard_name": "Keychron Q12",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x01D1",
+        "device_version": "1.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q12/ansi_encoder/info.json:10]:
```suggestion
    },
    "encoder": {
        "rotary": [
            {"pin_a": "A2", "pin_b": "C14"}
        ]
    },
```

File: keyboards/keychron/q12/ansi_encoder/config.h [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+#pragma once
+
+/* RGB Matrix Configuration */
+#define DRIVER_1_LED_TOTAL 56
+#define DRIVER_2_LED_TOTAL 46
+#define RGB_MATRIX_LED_COUNT (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+
+/* Encoder Configuration */
+#define ENCODERS_PAD_A { A2 }
+#define ENCODERS_PAD_B { C14 }
```
Comment (fauxpark) [inline keyboards/keychron/q12/ansi_encoder/config.h:26]:
```suggestion
```

File: keyboards/keychron/q12/iso_encoder/info.json [hunk 6]
```diff
@@ -0,0 +1,128 @@
+{
+    "keyboard_name": "Keychron Q12",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x01D3",
+        "device_version": "1.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q12/iso_encoder/info.json:10]:
```suggestion
    },
    "encoder": {
        "rotary": [
            {"pin_a": "A2", "pin_b": "C14"}
        ]
    },
```

File: keyboards/keychron/q12/iso_encoder/config.h [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+#pragma once
+
+/* RGB Matrix Configuration */
+#define DRIVER_1_LED_TOTAL 56
+#define DRIVER_2_LED_TOTAL 47
+#define RGB_MATRIX_LED_COUNT (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+
+/* Encoder Configuration */
+#define ENCODERS_PAD_A { A2 }
+#define ENCODERS_PAD_B { C14 }
```
Comment (fauxpark) [inline keyboards/keychron/q12/iso_encoder/config.h:26]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q12/ansi_encoder/info.json (hunk 4 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q12/ansi_encoder/info.json:10]: Thanks!

**Comment by keychron-dev** [comment]
@fauxpark please have a look of this. thanks

**Review by fauxpark — APPROVED** [review]

**Comment by keychron-dev** [comment]
can you merge this since we have two approvals on this? sorry about the message, we have users keep asking about this, thanks @fauxpark @drashna

**Pull request merged**